src/strategies/latin_hypercube.jl

```diff
@@ -75,32 +75,34 @@ function _create_bounds_and_dims(d,r)
                  transform(MLJBase.Scale,MLJBase.scale(r[i].scale),r[i].origin + r[i].unit)]))
             end
         else
-            push!(dims, LatinHypercubeSampling.Categorical(length(r[i].values), 1.0))
+            push!(dims_type, LatinHypercubeSampling.Categorical(length(r[i].values), 1.0))
             push!(bounds,Float64.([1,length(r[i].values)]))
         end
     end
-    return Tuple.(bounds), dims
+    return Tuple.(bounds), dims_type
 end
 
 function setup(tuning::LatinHypercube, model, r, verbosity)
     d = length(r)
-    bounds, dims = _create_bounds_and_dims(d, r)
+    bounds, dims_type = _create_bounds_and_dims_type(d, r)
     plan, _ = LatinHypercubeSampling.LHCoptim(tuning.n_max, d, tuning.nGenerations,
                     rng = tuning.rng,
                     popsize = tuning.popSize,
                     ntour = tuning.nTournament,
                     ptour = tuning.pTournament,
-                    dims = dims,
+                    dims = dims_type,
                     interSampleWeight = tuning.interSampleWeight,
                     periodic_ae = tuning.periodic_ae,
                     ae_power = tuning.ae_power)
     scaled_plan = LatinHypercubeSampling.scaleLHC(plan, bounds)
     for i = 1:size(scaled_plan,1)
         for j = 1:size(scaled_plan,2)
             if dims[j] isa LatinHypercubeSampling.Continuous
-                scaled_plan[i,j] = inverse_transform(MLJBase.Scale,
+                if r[j] isa MLJBase.NumericRange{Int,MLJBase.Bounded,Symbol}
```

Comment:
This is flawed. The type could be any subtype of `Integer` not just `Int=Int64`.  Please don't use `Int(floor...)` use `round(T, ...)` where `T <: Integer` is the type. Rather than use `if` statements, more Julian to use dispatch (more likely to be type-stable). You can see examples of this idea in my earlier comment.